### ISSUE TITLE
Arguments error 

### ISSUE THREAD

**Issue by jmadkins** [vlado/activerecord-cte#8]
I'm receiving a wrong number of arguments error running 0.1.5 with ActiveRecord 6.1.4. I'm happy to provide more information to assist.

```
activerecord (6.1.4) lib/active_record/relation/query_methods.rb:1136:in `build_arel'
activerecord-cte (0.1.5) lib/activerecord/cte/core_ext.rb:50:in `build_arel'
activerecord (6.1.4) lib/active_record/relation/query_methods.rb:1059:in `arel'
activerecord (6.1.4) lib/active_record/relation/calculations.rb:200:in `block in pluck'
activerecord (6.1.4) lib/active_record/relation.rb:861:in `skip_query_cache_if_necessary'
activerecord (6.1.4) lib/active_record/relation/calculations.rb:196:in `pluck'
activerecord (6.1.4) lib/active_record/schema_migration.rb:46:in `all_versions'
activerecord (6.1.4) lib/active_record/migration.rb:1109:in `get_all_versions'
activerecord (6.1.4) lib/active_record/migration.rb:1121:in `needs_migration?'
activerecord (6.1.4) lib/active_record/migration.rb:625:in `check_pending!'
activerecord (6.1.4) lib/active_record/migration.rb:590:in `block (2 levels) in call'
activesupport (6.1.4) lib/active_support/evented_file_update_checker.rb:59:in `execute'
activerecord (6.1.4) lib/active_record/migration.rb:595:in `block in call'
activerecord (6.1.4) lib/active_record/migration.rb:587:in `synchronize'
activerecord (6.1.4) lib/active_record/migration.rb:587:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
activesupport (6.1.4) lib/active_support/callbacks.rb:98:in `run_callbacks'
actionpack (6.1.4) lib/action_dispatch/middleware/callbacks.rb:26:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/actionable_exceptions.rb:18:in `call'
better_errors (2.9.1) lib/better_errors/middleware.rb:62:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/debug_exceptions.rb:29:in `call'
web-console (4.1.0) lib/web_console/middleware.rb:132:in `call_app'
web-console (4.1.0) lib/web_console/middleware.rb:19:in `block in call'
web-console (4.1.0) lib/web_console/middleware.rb:17:in `catch'
web-console (4.1.0) lib/web_console/middleware.rb:17:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
lograge (0.11.2) lib/lograge/rails_ext/rack/logger.rb:15:in `call_app'
railties (6.1.4) lib/rails/rack/logger.rb:26:in `block in call'
activesupport (6.1.4) lib/active_support/tagged_logging.rb:99:in `block in tagged'
activesupport (6.1.4) lib/active_support/tagged_logging.rb:37:in `tagged'
activesupport (6.1.4) lib/active_support/tagged_logging.rb:99:in `tagged'
railties (6.1.4) lib/rails/rack/logger.rb:26:in `call'
sprockets-rails (3.2.2) lib/sprockets/rails/quiet_assets.rb:13:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
request_store (1.5.0) lib/request_store/middleware.rb:19:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/request_id.rb:26:in `call'
rack (2.2.3) lib/rack/method_override.rb:24:in `call'
rack (2.2.3) lib/rack/runtime.rb:22:in `call'
activesupport (6.1.4) lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
sentry-ruby-core (4.6.1) lib/sentry/rack/capture_exceptions.rb:23:in `block in call'
sentry-ruby-core (4.6.1) lib/sentry/hub.rb:56:in `with_scope'
sentry-ruby-core (4.6.1) lib/sentry-ruby.rb:168:in `with_scope'
sentry-ruby-core (4.6.1) lib/sentry/rack/capture_exceptions.rb:14:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/static.rb:24:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/static.rb:24:in `call'
rack (2.2.3) lib/rack/sendfile.rb:110:in `call'
actionpack (6.1.4) lib/action_dispatch/middleware/host_authorization.rb:98:in `call'
webpacker (5.4.0) lib/webpacker/dev_server_proxy.rb:25:in `perform_request'
rack-proxy (0.7.0) lib/rack/proxy.rb:63:in `call'
railties (6.1.4) lib/rails/engine.rb:539:in `call'
puma (5.3.2) lib/puma/configuration.rb:249:in `call'
puma (5.3.2) lib/puma/request.rb:77:in `block in handle_request'
puma (5.3.2) lib/puma/thread_pool.rb:338:in `with_force_shutdown'
puma (5.3.2) lib/puma/request.rb:76:in `handle_request'
puma (5.3.2) lib/puma/server.rb:438:in `process_client'
puma (5.3.2) lib/puma/thread_pool.rb:145:in `block in spawn_thread' 
```

**Comment by vlado** [comment]
Thanks for reporting @jmadkins. It would be great if you could provide more info (exact error message, ...) or some guide how to reproduce the error.

**Comment by jmadkins** [comment]
Sorry for the late reply. I stopped work on adding the gem but will pick back up tonight and get an exact error.

**Comment by shanet** [comment]
I have this issue as well with Rails 6.1.4. I believe it comes from the 6.1.4 fix here: https://github.com/vlado/activerecord-cte/pull/6/files#diff-0cfea5c52d07452497a8c718d66f591f16d71457c596d48f004707bd018df025R49.

Namely, using both `*args` and `*kwargs` means both of those are passed to `super` which causes the following error:

```
ArgumentError:
  wrong number of arguments (given 2, expected 0..1)
gems/activerecord-6.1.4/lib/active_record/relation/query_methods.rb:1136:in `build_arel'
bundler/gems/activerecord-cte-2fe38fbd8fd9/lib/activerecord/cte/core_ext.rb:50:in `build_arel'
gems/activerecord-6.1.4/lib/active_record/relation/query_methods.rb:1059:in `arel'
gems/activerecord-6.1.4/lib/active_record/relation/calculations.rb:200:in `block in pluck'
gems/activerecord-6.1.4/lib/active_record/relation.rb:861:in `skip_query_cache_if_necessary'
gems/activerecord-6.1.4/lib/active_record/relation/calculations.rb:196:in `pluck'
gems/activerecord-6.1.4/lib/active_record/internal_metadata.rb:39:in `[]'
gems/activerecord-6.1.4/lib/active_record/tasks/database_tasks.rb:357:in `schema_up_to_date?'
gems/activerecord-6.1.4/lib/active_record/migration.rb:633:in `block in load_schema_if_pending!'
gems/activerecord-6.1.4/lib/active_record/migration.rb:632:in `all?'
gems/activerecord-6.1.4/lib/active_record/migration.rb:632:in `load_schema_if_pending!'
gems/activerecord-6.1.4/lib/active_record/migration.rb:653:in `block in maintain_test_schema!'
gems/activerecord-6.1.4/lib/active_record/migration.rb:906:in `suppress_messages'
gems/activerecord-6.1.4/lib/active_record/migration.rb:658:in `method_missing'
gems/activerecord-6.1.4/lib/active_record/migration.rb:653:in `maintain_test_schema!'
```

The fix would be to change the `build_arel` method to the following (removing the `kwargs` argument):

```
    def build_arel(*args)
      arel = super
      build_with(arel) if @values[:with]
      arel
    end
```

**Comment by vlado** [comment]
Thanks @shanet, that makes sense.

I had an operation on Wednesday and will not be able to do any work for at least a week. Can you submit a PR with the fix maybe?

**Comment by vlado** [comment]
@shanet & @jmadkins Would like to write failing test before fixing this. I quickly tried to reproduce it but had no luck. Tried with Ruby 3.0.2 and 2.7.3 and it worked without problems. Which version of Ruby are you using?

**Comment by jmadkins** [comment]
@vlado I'm using ruby 2.6.3 and rails 6.1.4

**Comment by shanet** [comment]
Ruby 2.6 here as well. I'm guessing the difference in behavior between 2.6 and 3.0 is a result of the changes to how keyword args are handled in Ruby 2.7.

**Comment by vlado** [comment]
Should be fixed in verison `0.2.0`.
